broker.go

```diff
@@ -452,14 +452,8 @@ func (b *Broker) coordinatorConnection(consumerGroup string) (conn *connection,
 
 // offset will return offset value for given partition. Use timems to specify
 // which offset value should be returned.
-func (b *Broker) offset(topic string, partition int32, timems int64) (offset int64, err error) {
-	conn, err := b.leaderConnection(topic, partition)
-	if err != nil {
-		return 0, err
-	}
-	defer func() { go b.conns.Idle(conn) }()
-
-	resp, err := conn.Offset(&proto.OffsetReq{
+func (b *Broker) offset(topic string, partition int32, timems int64) (int64, error) {
```

Comment:
Nit: What's the motivation for the name `timems`?  The name makes me think that it's a timestamp in milliseconds, but it looks like it's being used as an enum constant of some sort.